dynamodb/dynamo_lock_table.go

```diff
@@ -99,7 +99,48 @@ func CreateLockTable(tableName string, readCapacityUnits int, writeCapacityUnits
 		}
 	}
 
-	return waitForTableToBeActive(tableName, client, MAX_RETRIES_WAITING_FOR_TABLE_TO_BE_ACTIVE, SLEEP_BETWEEN_TABLE_STATUS_CHECKS, terragruntOptions)
+	err = waitForTableToBeActive(tableName, client, MAX_RETRIES_WAITING_FOR_TABLE_TO_BE_ACTIVE, SLEEP_BETWEEN_TABLE_STATUS_CHECKS, terragruntOptions)
+
+	if err != nil {
+		return err
+	}
+
+	if createTableOutput != nil && createTableOutput.TableDescription != nil && createTableOutput.TableDescription.TableArn != nil {
+		// Do not tag in case somebody else had created the table
```

Comment:
Oh, that's a very interesting point. Checking `createTableOutput` makes sense; why check the description field?

Reply:
That part crashed in a test, so I guessed it was necessary as well. Just to be on the safe side...

Reply:
Crashed in a test? Because you didn't check description? Do you have the stack trace?

Reply:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xa0 pc=0x132babc]

goroutine 190 [running]:
github.com/gruntwork-io/terragrunt/dynamodb.CreateLockTable(0xc420014cc0, 0x16, 0x0, 0x1, 0x1, 0xc42000e0c8, 0xc4201e0000, 0x0, 0x0)
        /Users/bernhardmercateo/go/src/github.com/gruntwork-io/terragrunt/dynamodb/dynamo_lock_table.go:108 +0x4dc
github.com/gruntwork-io/terragrunt/dynamodb.CreateLockTableIfNecessary(0xc420014cc0, 0x16, 0x0, 0xc42000e0c8, 0xc4201e0000, 0x0, 0x0)
        /Users/bernhardmercateo/go/src/github.com/gruntwork-io/terragrunt/dynamodb/dynamo_lock_table.go:48 +0x161
github.com/gruntwork-io/terragrunt/dynamodb.TestCreateLockTableConcurrency.func1(0xc420020f80, 0xc420014cc0, 0x16, 0xc42000e0c8, 0xc4201e0000, 0xc4201c43c0)
        /Users/bernhardmercateo/go/src/github.com/gruntwork-io/terragrunt/dynamodb/dynamo_lock_table_test.go:47 +0x89
created by github.com/gruntwork-io/terragrunt/dynamodb.TestCreateLockTableConcurrency
        /Users/bernhardmercateo/go/src/github.com/gruntwork-io/terragrunt/dynamodb/dynamo_lock_table_test.go:45 +0x1b9
exit status 2
```

Hm, you think if createTableOutput is not nil, I can rely on the other two fields to be filled and don't need to check?

Reply:
Oh, I misunderstood `TableDescription`; I thought it was an optional human-readable description, rather than a struct used to return data about the table such as its ARN. OK, this is probably fine then.